crates/storage/src/structured_storage/coins.rs

```diff
@@ -11,12 +10,115 @@ use crate::{
     tables::Coins,
 };
 
-impl TableWithBlueprint for Coins {
-    type Blueprint = Plain<Primitive<34>, Postcard>;
-    type Column = Column;
+#[cfg(not(feature = "global-state-root"))]
+mod plain {
```

Comment:
We don't need to integrate it into the `fuel-core-storage`. It should be a new stance alone crate that can accept the block as an input and will create Merkle roots over data from there=D

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2553/commits/13521ac08c857a093836a9bbf02f43986099a89c (by you ;) )